src/main/java/org/littleshoot/proxy/impl/ClientToProxyConnection.java

```diff
@@ -321,6 +338,16 @@ private ConnectionState doReadHTTPInitial(HttpRequest httpRequest) {
         }
     }
 
+    private boolean isProxyRequested() {
+        // HTTPS requests have uri without http scheme too
+        if (currentRequest.getMethod() == HttpMethod.CONNECT || sslEngine != null) {
+            return false;
+        }
+        // direct requests to the proxy have the path only without a scheme
```

Comment:
Is this statement accurate? If the proxy were running at (for example) 192.168.100.100, and I made an HTTP request to http://192.168.100.100, wouldn't request.getUri() return "http://192.168.100.100"?